examples/apps/simple_wallet/workload/simple_wallet_execute_io.cpp

```diff
@@ -0,0 +1,215 @@
+/* Copyright 2019 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include <string>
+#include <string.h>
+
+#include "types.h"
+#include "jsonvalue.h"
+#include "json_utils.h"
+#include "crypto_utils.h"
+#include "error.h"
+#include "tcf_error.h"
+#include "enclave_utils.h"
+#include <parson.h>
+
+#include "simple_wallet_execute_io.h"
+#include "file_io.h"
+//#include "workload_crypto_utils.h"
+
+#define LEDGER_FILE "/tmp/account_ledger.json"
+
+SimpleWalletLedger* SimpleWalletLedger::instance = 0;
+
+SimpleWalletIoExecutor::SimpleWalletIoExecutor() {}
+
+// Create encryption key used for encrypting or decrypting
+// ledger having all wallet details
+bool SimpleWalletIoExecutor::CreateLedgerEncryptionKey() {
+    std::string enc_key = tcf::crypto::CreateHexEncodedEncryptionKey();
+    if (enc_key.empty()) {
+        return false;
+    }
+    SimpleWalletLedger* ledger = SimpleWalletLedger::GetInstance();
+    ledger->ledger_encryption_key = enc_key;
```

Comment:
One options is to persist encryption key by sgx sealing and unsealing during restart scenario.
But we can't ensure replay protection since latest sgx versions doesn't support monotonic counter.

I will raise a separate PR to take care of persisting key on the file.

Reply:
I have raised below issue to track.
https://github.com/hyperledger/avalon/issues/198